tesla_http_proxy/DOCS.md

```diff
@@ -14,38 +14,103 @@ servers: nginx_proxy/*.conf
+Request application access at [developer.tesla.com](https://developer.tesla.com). My request was approved immediately but YMMV. This is currently free but it's possible they will monetize it in the future. You will need to provide the following information:
 
 - Name of your legal entity (first and last name is fine)
 - App Name, Description, Purpose (can be anything)
 - Allow all scopes
 - **OAuth Grant Type**: Authorization code and machine-to-machine
-- **Allowed Origin**: The FQDN where you are hosting the public key.  Must be lowercase, e.g. `https://tesla.example.com`
+- **Allowed Origin**: The FQDN where you are hosting the public key. Must be lowercase, e.g. `https://tesla.example.com`
 - **Redirect URI**: Append `/callback` to the FQDN, e.g. `https://tesla.example.com/callback`
 
-Tesla will provide a Client ID and Client Secret.  Enter these in addon configuration and then Start it again.  Now the `regenerate auth` setting will be automatically disabled.
+Tesla will provide a Client ID and Client Secret. Enter these in addon configuration and then Start it again. Now the `regenerate auth` setting will be automatically disabled.
 
 Use the [Tesla Auth app](https://apps.apple.com/us/app/auth-app-for-tesla/id1552058613) to obtain a refresh token, by entering your Callback URL, Client ID, and Client Secret on the Fleet API page. This will open a webpage that asks for your Tesla account credentials.
 
 > [!TIP]
 > The first time you request a refresh token, it will also prompt to authorize your Client ID to access your Tesla account. Allow all scopes.
 
-Using the Home Assistant iOS app, open the Addon Web UI and click **Enroll public key in your vehicle**.  This should launch the Tesla app where it prompts for approval to "allow third-party access to your vehicle".  If you have multiple vehicles, you'll need to do this on each of them.
+Using the Home Assistant iOS app, open the Addon Web UI and click **Enroll public key in your vehicle**. This should launch the Tesla app where it prompts for approval to "allow third-party access to your vehicle". If you have multiple vehicles, you'll need to do this on each of them.
+
 > [!NOTE]
 > Your Tesla app must already be key-paired with the car.
 
 Configure the [Tesla integration](https://github.com/alandtse/tesla) to use this proxy. It should pre-fill the Client ID, URL, and certificate for you by reading them from this addon.
 
+## Specific setup guides
+
+### Cloudflare
+
+Setting up Tesla HTTP Proxy through cloudflare involves the following steps (all the steps are made with the available addons in home assistant, nothing external. Also it avoids the necessity to open ports on the router):
+
+1. Install **Cloudflared** addon. Then configure like this and start the addon.
+
+```yml
+external_hostname: ha.example.com
+additional_hosts:
+  - hostname: tesla.example.com
+    service: https://homeassistant-internal.example.com
+```
```

Comment:
Does the add-on use the CF API to create the tunnel OR does the end user needs to configure that as well?

Reply:
The addon uses CF API, no other input needed to create the tunnel